File: app/src/main/java/nz/co/codebros/quakesnz/module/QuakeListModule.java
package nz.co.codebros.quakesnz.module;

import dagger.Module;
import dagger.Provides;
import nz.co.codebros.quakesnz.GeonetService;
import nz.co.codebros.quakesnz.interactor.GetFeaturesInteractor;
import nz.co.codebros.quakesnz.presenter.QuakeListPresenter;
import nz.co.codebros.quakesnz.view.QuakeListView;

/**
 * Created by leandro on 9/07/15.
 */
@Module
public class QuakeListModule {
    private QuakeListView view;

    public QuakeListModule(QuakeListView view) {
        this.view = view;
    }

    @Provides
    public GetFeaturesInteractor provideInteractor(GeonetService service) {
        return new GetFeaturesInteractor(service);
    }

    @Provides
    public QuakeListPresenter providePresenter(GetFeaturesInteractor interactor) {
        return new QuakeListPresenter(view, interactor);
    }
}

File: app/src/main/java/nz/co/codebros/quakesnz/list/QuakeListModule.kt
package nz.co.codebros.quakesnz.list

import dagger.Binds
import dagger.Module
import nz.co.codebros.quakesnz.interactor.LoadFeaturesInteractor
import nz.co.codebros.quakesnz.interactor.LoadFeaturesInteractorImpl

/**
 * Created by Leandro on 27/11/2017.
 */
@Module
abstract class QuakeListModule {

    @Binds
    abstract fun loadFeaturesInteractor(
            loadFeaturesInteractorImpl: LoadFeaturesInteractorImpl
    ): LoadFeaturesInteractor
}